### PR TITLE
ref(filters): Move generic filter checks to event::filter

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Follow-up to https://github.com/getsentry/relay/pull/3161#discussion_r1508884588.

This PR moves the check whether Relay supports generic filters from the processor to `event::filter`, which returns an enum of what Relay should do with the event. There are no functional changes.

#skip-changelog

### CODE REVIEW

File: relay-server/src/services/processor/event.rs [hunk 1]
```diff
@@ -271,15 +271,35 @@ pub fn finalize<G: EventProcessing>(
     Ok(())
 }
 
+/// Decision for applying some filters that don't drop the event.
```
Comment (iker-barriocanal) [inline relay-server/src/services/processor/event.rs]: Bikeshedding a bit: I don't really like "decision" here, but the alternatives "result" and "outcome" have other meanings that may generate some confusion. Open to suggestions.

File: relay-server/src/services/processor/event.rs [hunk 2]
```diff
@@ -271,15 +271,35 @@ pub fn finalize<G: EventProcessing>(
...
+///
+/// The enum represents either the success of running all filters and keeping
+/// the event, [`FiltersApplied::Ok`], or not running all the filters because
+/// some are unsupported, [`FiltersApplied::Unsupported`].
+///
+/// If there are unsuppported filters, Relay should forward the event upstream
+/// so that a more up-to-date Relay can apply filters appropriately. Actions
+/// that depend on the outcome of event filtering, such as metric extraction,
+/// should be postponed until a filtering decision is made.
+pub enum FiltersApplied {
```
Comment (Dav1dde) [inline relay-server/src/services/processor/event.rs]: What about:

```rs
enum FilterStatus {
    Ok,
    Unsupported,
}
```

Comment (Dav1dde) [inline relay-server/src/services/processor/event.rs]:
```suggestion
#[must_use]
pub enum FiltersApplied {
```

**Review by Dav1dde — APPROVED** [review]
Thanks for following up on this.

**Pull request merged**